Ardupilog.m

```diff
@@ -228,13 +199,48 @@
         end
             
         function [] = findMsgStart(obj)
-        % Read bytes from the file till the message-start character is found (dec=163,hex=A3)
-            b1 = fread(obj.fileID, 1, 'uint8', 0, 'l');
-            b2 = fread(obj.fileID, 1, 'uint8', 0, 'l');            
-            while (feof(obj.fileID)==0) && (b1 ~= obj.header1) && (b2 ~= obj.header2)
-                disp(['Warning: Trashing byte from log! hex=', dec2hex(b1,2),' dec=', b1,' char=',char(b1)])
-                b1 = b2; % move 2nd byte to 1st pos, old b1 is trashed
-                b2 = fread(obj.fileID, 1, 'uint8', 0, 'l'); % read new byte into b2
+        % Read bytes from the file till the message-start character is found (dec=163,hex=A3).
+        % (Leave read_ndx pointing BEFORE the header.)
+            
+            if all(obj.log_data((1:length(obj.header))+obj.log_data_read_ndx) == obj.header)
+                % Header is next set of bytes, don't trash anything, and continue
+            elseif (numel(obj.log_data) - obj.log_data_read_ndx) < length(obj.header)
+                % Log ends mid-header. No need to warn user, just read incomplete bit and exit silently
+                nbytes_remaining = length(obj.log_data)-obj.log_data_read_ndx;
+                trash_silent_bytes = obj.readLogData(nbytes_remaining);
+            else
+                % Header is NOT next 2 bytes... something went wrong.
+                disp(['Warning: something went wrong, either in the ' ...
+                      'log formation or the log processing.'])
+                disp(['Line# ', num2str(obj.lastLineNum), ' ended without a header beginning a new message'])
+                hdr_find = strfind(obj.log_data((obj.log_data_read_ndx+1):end), obj.header);
+                if isempty(hdr_find)
+                    disp('No more headers exist in log')
+                    % There are no more valid headers in the log, it is probably done
+                    % Trash rest of log and move read_ndx to end, signaling finish
+                    nbytes_remaining = length(obj.log_data)-obj.log_data_read_ndx;
+                    trash_bytes = obj.readLogData(nbytes_remaining);
+                else
+                    % Trash part of log and resume
+                    num_to_trash = hdr_find(1)-1;
+                    trash_bytes = obj.readLogData(num_to_trash);
```

Comment:
From above, `hdr_find = strfind(obj.log_data((obj.log_data_read_ndx+1):end), obj.header);`

hdr_find begins counting at the `obj.log_data_read_ndx + 1` position, so in your example hdr_find would = 6, corresponding to actual log_data position 705. Since hdr_find = 6, then `num_to_trash = hdr_find(1) - 1;` would be 5, trashing bytes 700,701,702,703,704 from the log.